### DISCUSSION TITLE
Move a source to the current location of another defined source.

### DISCUSSION THREAD

**Discussion by Grumtastic** [exeldro/obs-move-transition#157] [Ideas]
Use case scenario.

I move my webcam to different locations on the screen based on the game I am playing (sometimes manually).
When I do a shout-out I Move a video clip from their stream onto a fixed position on the screen for a short period, then Move it off again.
I want to Move it to sit over my webcam image, so it doesn't obscure too much of the gameplay. This is difficult as I won't always know where my webcam is.

The obvious solution to this would be to add a dropdown in the Transform section of Move Source to choose another source. The Move filter will then grab the current transform data from that source and use that as a target instead of the manual one.

Expanding on this would be choosing which transform parameters to copy. (position, dimensions, both).

What do you think?

**Comment by exeldro** [comment]
I like the idea, you got me thinking how to make it easy for users to select what part of the transform to copy. Maybe make it a separate filter.

**Reply by Grumtastic** [reply]
I'm glad you like my idea!

It would make the existing Move Source command a bit unwieldy, so a separate filter probably makes more sense as this gives more scope and expandability.

I was thinking afterwards that another thing people might want to do is centre the moving source to match the centre of the target source (if the sources are different sizes).

I definitely think that this simple feature has far-reaching possibilities.

One possible snag would be how to handle if the target source is also being moved at the same time. Personally, I would want to use the original position.
